tools/SourceKit/lib/Core/Context.cpp

```diff
@@ -16,9 +16,11 @@
 
 using namespace SourceKit;
 
-SourceKit::Context::Context(StringRef RuntimeLibPath)
+SourceKit::Context::Context(StringRef RuntimeLibPath,
+    		llvm::function_ref<
+    			std::unique_ptr<LangSupport>(Context &)> LangSupportFactoryFn)
   : RuntimeLibPath(RuntimeLibPath),
-    SwiftLang(LangSupport::createSwiftLangSupport(*this)),
+    SwiftLang(LangSupportFactoryFn(*this)),
     NotificationCtr(new NotificationCenter()) {
 }
 
```

Comment:
Oh yes, good call! That crossed my mind at one point as well, but then I forgot again.